### ISSUE TITLE
`bun wiptest` should return a non-zero exit code when tests can't run

### ISSUE THREAD

**Issue by Nick-Mazuk** [oven-sh/bun#1935]
### What version of Bun is running?

0.5.1

### What platform is your computer?

Darwin 22.2.0 arm64 arm

### What steps can reproduce the bug?

Let's say you have the following test file. This test never runs because the test has invalid syntax.

```js
import { expect, it } from "bun:test"

it('fails', () => {
    expect(true).toBe(true)l
})
```

Yet when you run `bun wiptest`, it has the exit code of 0.

### What is the expected behavior?

Syntactically invalid tests should produce a non-zero exit code.

Most platforms (such as GitHub actions) assume that an exit code of 0 means success, and non-zero exit codes mean an error. So when I try and use `bun wiptest` in GitHub actions with tests that never run because they cannot be compiled, the action passes.

### What do you see instead?

`bun wiptest` reports the syntax error, but since none of the expectations failed (because none ran), it exits with a code of 0.

### Additional information

In my opinion, for other runtime errors (such as module not found errors), a non-zero exit code should be returned as well.

**Comment by Jarred-Sumner** [comment]
its definitely supposed to

hm

**Comment by Nick-Mazuk** [comment]
Sorry, debugged this a bit further and my initial bug report was a bit misleading.

The file I was testing had a syntax error so the test never ran. When I simplified the code even just a little bit to create this bug report, I accidentally removed that syntax error.

I was able to validate that with syntactically correct code, `bun wiptest` will exit with an exit code of `1`.

The correct version of this bug report should be that if tests do not run due to bugs in the code itself (e.g., syntax errors, potentially even import-not-found errors), there should be a non-zero exit code.

In my opinion, an exit code of 0 should only be returned if all the code runs and all the tests pass. If code cannot run because it's invalid, it should still return a non-zero exit code.

Will update the original question.
